src/base/static/js/routes.js

```diff
@@ -36,97 +76,193 @@ Shareabouts.Util = Util;
+      );
 
-      this.loading = false;
-    },
+      //      // Load the default page when there is no page already in the url
+      //      if (Backbone.history.getFragment() === "") {
+      //        const startPageConfig = _.findWhere(options.navBarConfig, {
+      //          start_page: true,
+      //        });
+      //
+      //        if (
+      //          startPageConfig &&
+      //          startPageConfig.url &&
+      //          // don't route to the start page on small screens
+      //          $(window).width() > (startPageConfig.show_above_width || 960)
+      //        ) {
+      //          this.navigate(startPageConfig.url, { trigger: true });
+      //        }
+      //      }
 
-    getCurrentPath: function() {
-      var root = Backbone.history.root,
-        fragment = Backbone.history.fragment;
-      return root + fragment;
+      // Start tracking routing history.
+      Backbone.history.start({ pushState: true });
     },
 
+    //  getCurrentPath: function() {
+    //    var root = Backbone.history.root,
+    //      fragment = Backbone.history.fragment;
+    //    return root + fragment;
+    //  },
+
     viewMap: function(zoom, lat, lng) {
       recordGoogleAnalyticsHit("/");
-      this.appView.viewMap(parseInt(zoom), parseFloat(lat), parseFloat(lng));
+      this.store.dispatch(updateCurrentTemplate("map"));
+      this.store.dispatch(updateUIVisibility("contentPanel", false));
+      this.store.dispatch(updateUIVisibility("spotlightMask", false));
+      this.store.dispatch(updateUIVisibility("mapCenterpoint", false));
+      this.store.dispatch(updateActivePage(null));
+      this.store.dispatch(updateContentPanelComponent(null));
+      this.store.dispatch(updateAddPlaceButtonVisibility(true));
+      zoom &&
+        lat &&
+        lng &&
+        this.store.dispatch(
+          updateMapViewport({
+            zoom: parseFloat(zoom),
+            lat: parseFloat(lat),
+            lng: parseFloat(lng),
+          }),
+        );
     },
 
     viewDashboard: function() {
       recordGoogleAnalyticsHit("/dashboard");
-      this.appView.viewDashboard();
+      this.store.dispatch(updateCurrentTemplate("dashboard"));
     },
 
     newPlace: function() {
-      this.appView.newPlace();
+      recordGoogleAnalyticsHit("/new");
+      this.store.dispatch(updateContentPanelComponent("InputForm"));
+      this.store.dispatch(updateAddPlaceButtonVisibility(false));
+      this.store.dispatch(updateUIVisibility("contentPanel", true));
     },
 
-    viewPlace: function(placeSlug, placeId, responseId) {
-      recordGoogleAnalyticsHit(`/${placeSlug}/${placeId}`);
-      this.appView.viewPlace({
-        placeSlug: placeSlug,
-        placeId: parseInt(placeId),
-        responseId: parseInt(responseId),
-      });
+    // TODO: responseId
+    viewPlace: async function(clientSlug, placeId, responseId) {
+      if (!placeExists(this.store.getState(), placeId)) {
```

Comment:
Here is where I reinstated the "direct load to place" functionality (i.e. if you load the page at something like `localhost:8000/place/1234` that Place's data will be fetched immediately).

This adds a little more complexity to the `viewPlace` route and the `places` duck, but I think the usability win is worth it.